herd/semExtra.ml

```diff
@@ -497,6 +499,34 @@ type concrete =
 
     type pp_barrier = { barrier:barrier ; pp:string; }
 
+  let patch_variant test variant cache_type =
+    let dic_pred, idc_pred =
+        let open CacheType in
+          match cache_type with
+          | None ->
+              (fun _ -> false), (fun _ -> false)
+          | Some cache_type ->
+              cache_type.dic, cache_type.idc in
+    let procs =
+      List.map (fun (p,_,_) -> p) (test.Test_herd.start_points) in
+    let all_dic, all_idc, any_dic, any_idc = List.fold_left
+        (fun (_all_dic,_all_idc,_any_dic,_any_idc) p -> (
+          _all_dic && (dic_pred p),
+          _all_idc && (idc_pred p),
+          _any_dic || (dic_pred p),
+          _any_idc || (idc_pred p)
+        ))
+        (true, true, false, false)
+        procs in
```

Comment:
I tend to find using stdlib utils clearer:

```suggestion
    let all_dic = List.for_all dic_pred procs
    and all_idc = List.for_all idc_pred procs
    and any_dic = List.exists dic_pred procs
    and any_idc = List.exists dic_pred procs
    in
```

Reply:
Ah, of course -- thanks for this.